api/v1alpha1/virtualmachine_types.go

```diff
@@ -106,6 +106,16 @@ const (
 	// marked userConfigurable and already present in either OVF Properties of a VirtualMachineImage
 	// or as vApp properties on an existing VM or VMTX will be set, all others will be ignored.
 	VirtualMachineMetadataOvfEnvTransport VirtualMachineMetadataTransport = "OvfEnv"
+
+	// VirtualMachineMetadataCloudInitTransport indicates the data in
+	// the ConfigMap resource specified by
+	// VirtualMachineMetadata.ConfigMapName is cloud-init userdata.
+	//
+	// Please note that, despite the name, VirtualMachineMetadata has no
+	// relationship to cloud-init instance metadata.
+	//
+	// For more information, please refer cloud-init's official document.
```

Comment:
document -> documentation, or just provide the link as in @akutz suggestion.

Do we say what the CM key is used for the data?

Reply:
Even Andrew suggested not to the link because it is some .html link. This key is used for cloud-init userdata.

Reply:
Seems prudent though to document the _name_ of the expected key in the CM, because we are expecting a specific one.

Reply:
Made changes as per our discussion.